File: 0.1.0/network/sequential.py
import numpy as np
import gc
from layer.dense import Dense

class Hidden_Input_Layer:

    def __init__(self):
        self.output = []

class Sequential:

    def __init__(self):
        self.layers = []
        HIL = Hidden_Input_Layer()
        self.layers.append(HIL)

    def add_layer(self, layer):
        self.layers.append(layer)

    def run(self, input):
        self.layers[0].output = input
        for i in range(1, len(self.layers)-1):
            self.layers[i].run(self.layers[i-1].output)
        return self.layers[len(self.layers)-2].output

File: 0.2.2/network/sequential.jl
module sequential
    mutable struct Sequential
        layers::Array{Any}
        add_layer::Any
        activator::Any
        num_layer::Int64

        function Sequential()
            new(Any[Hidden_Input_Layer(Float32[])], Sequential_add_layer, activate_Sequential, 0)
        end
    end

    mutable struct Hidden_Input_Layer
        output::Array{Float32}
    end
    mutable struct Hidden_Output_Layer
        propagation_units::Array{Float32}
    end

    function Sequential_add_layer(model::Sequential, layer::Any)
        push!(model.layers, layer)
        model.num_layer += 1
    end

    function activate_Sequential(model::Sequential, data::Array{Float32})
        model.layers[1].output = data
        if length(model.layers)==model.num_layer+1
            push!(model.layers, Hidden_Output_Layer(Float32[]))
        end
        for i in 2:length(model.layers)-1
            model.layers[i].activator(model.layers[i], model.layers[i-1].output)
        end
        return model.layers[end-1].output
    end
end
